File: doc-examples/jdbc-spring-template-example-java/src/main/java/example/Book.java
package example;

import org.jspecify.annotations.Nullable;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

// tag::book[]
@MappedEntity
public record Book(
    @Id @GeneratedValue @Nullable Long id,
    String title,
    int pages
) { }
// end::book[]

File: doc-examples/jdbc-spring-template-example-kotlin/src/main/kotlin/example/Book.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity

// tag::book[]
@MappedEntity
data class Book(
    @field:Id @field:GeneratedValue val id: Long?,
    val title: String,
    val pages: Int
)
// end::book[]
